src/predictor/cpu_predictor.cc

```diff
@@ -274,13 +274,23 @@ float FillNodeMeanValues(RegTree const *tree, bst_node_t nidx, std::vector<float
   return result;
 }
 
-void FillNodeMeanValues(RegTree const* tree, std::vector<float>* mean_values) {
+void FillNodeMeanValues(RegTree const* tree, std::vector<float>* mean_values, bool approximate) {
   size_t num_nodes = tree->param.num_nodes;
   if (mean_values->size() == num_nodes) {
     return;
   }
   mean_values->resize(num_nodes);
-  FillNodeMeanValues(tree, 0, mean_values);
+
+  if (approximate) {
+    // prediction value as if the node is a leaf
+    auto &node_mean_values = *mean_values;
+    for (bst_node_t nidx = 0; nidx < num_nodes; ++nidx) {
+      node_mean_values[nidx] = tree->Stat(nidx).base_weight;
```

Comment:
I think `tree->Stat(nidx).base_weight` is not adjusted for learning rate. So this prediction would be much too large.

Reply:
That's true, but unfortunately, we cannot access the training parameters (i.e. `learning_rate`) here. Do you have a better idea than the following hack?

https://github.com/dmlc/xgboost/pull/2003/commits/87d96d2c13663122346216c8eba0829dfdcbf175#diff-0c9f721872945f814f6d5e9ade9cd184b7e1882ce6a138f183f5af07c8520c65R596-R600